DEVELOPER.md

```diff
@@ -0,0 +1,51 @@
+# Developer Notes
```

Comment:
Since we also need a CONTRIBUTING.md (how, legalities, attribution, etc, see librdkafka) it might be better to move this developer stuff in there as well.

Reply:
sounds good. deferred to future iteration :-).